llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -8100,6 +8100,13 @@ Instruction *InstCombinerImpl::visitFCmpInst(FCmpInst &I) {
         return new FCmpInst(I.getSwappedPredicate(), X, NegC, "", &I);
   }
 
+  // fcmp (fadd X, 0.0), Y --> fcmp X, Y
+  if (match(Op0, m_FAdd(m_Value(X), m_APFloat(C))) && C->isZero()) {
+    if (match(Op1, m_Value(Y))) {
+      return new FCmpInst(Pred, X, Y, "", &I);
+    }
+  }
```

Comment:
```suggestion
  if (match(Op0, m_FAdd(m_Value(X), m_AnyZeroFP())))
      return new FCmpInst(Pred, X, Op1, "", &I);
  if (match(Op1, m_FAdd(m_Value(X), m_AnyZeroFP())))
      return new FCmpInst(Pred, Op0, X, "", &I);
```

1. Add a test for `fcmp X, (fadd Y, 0.0)`
2. Add a vector test for `fcmp (fadd X, <0.0, -0.0>), Y`
3. Add a test for fast-math flag preservation

Reply:
> 1. Add a test for `fcmp X, (fadd Y, 0.0)`
> 
>     2. Add a vector test for `fcmp (fadd X, <0.0, -0.0>), Y`
> 
>     3. Add a test for fast-math flag preservation

Should I write a separate test case for each type of `fcmp` instruction (`ugt`, `ogt`, etc) for each of these? Or is one test case sufficient for each?

Reply:
Just one test each is fine.